server/controllers/admin.py

```diff
@@ -675,11 +676,14 @@ def autograde(cid, aid):
         return abort(404)
     form = forms.CSRFForm()
     if form.validate_on_submit():
-        try:
-            autograder.autograde_assignment(assign)
-            flash('Submitted to the autograder', 'success')
-        except ValueError as e:
-            flash(str(e), 'error')
+        job = jobs.enqueue_job(
+            autograder.autograde_assignment,
+            description='Autograde {}'.format(assign.display_name),
+            timeout=2 * 60 * 60,  # 2 hours
```

Comment:
I don't have an issue with the overall job timeout being 2 hours -  My comment is more directed at `QUEUED_TIMEOUT`